compiler/tools/suggest.nim

```diff
@@ -477,6 +481,115 @@ proc findTrackedSym*(g: ModuleGraph;): PSym =
     # xxx: the node finding should be specialized per symbol kind
     result = findTrackedNode(m.ast, g.config.m.trackPos)
 
+proc getSymNode(node: ParsedNode): ParsedNode =
+  result = node
+  if result.kind == pnkPostfix:
+    result = result[^1]
+  elif result.kind == pnkPragmaExpr:
+    result = getSymNode(result[0])
```

Comment:
```suggestion
  case node.kind
  of pnkPostfix:    node[^1]
  of pnkPragmaExpr: getSymNode(node[0])
  of pnkIdent:      node
  else:             unreachable(node.kind)
```
Multiple things of note:
1. expression style is preferred
2. where possible `case` statements/expressions are preferred over `if` statements/expressions. Single line branches can be used if the bodies are short
3. `case` statements should be exhaustive whenever possible
4. procedures like this one should be very in strict in what they accept as input

Being very strict about the inputs and treating violations as hard internal errors helps detect violations of invariants early on.